src/Document/FilterElement.ts

```diff
@@ -41,13 +47,9 @@ export default class FilterElement extends Element {
 			return;
 		}
 
-		const x = Math.floor(boundingBox.x1);
-		const y = Math.floor(boundingBox.y1);
-		// temporarily remove filter to avoid recursion
-		const filter = element.getStyle('filter').getString();
-
-		element.getStyle('filter').setValue('');
-
+		const x = Math.floor(boundingBox.x);
+		const y = Math.floor(boundingBox.y);
+		const ignoredStyles = this.removeStyles(element, FilterElement.ignoreStyles);
```

Comment:
instead of ignoring/restoring, we can potentially make the `style` method only crawl up the tree up to a specific ancestor when it's traversing https://github.com/canvg/canvg/blob/master/src/Document/Element.ts#L152

Reply:
To render this element in subcanvas, we must remove certain styles.

https://github.com/canvg/canvg/pull/978/files#diff-295489ec40ac69f2e4c53d1e249bf114R69